src/main/ml-modules/root/test/test-controller.xqy

```diff
@@ -147,6 +192,29 @@ declare function run($suite as xs:string, $name as xs:string, $module, $run-tear
 };
 
 
+declare function format($result as element(), $format as xs:string, $suite as xs:string)
+{
+    if ($format eq "junit") then
+        format-junit($suite)
+    else
+        let $format-uris := helper:list-from-database($db-id, $root, (), 'formats')
```

Comment:
The code above checked whether `$db-id` is 0, and if so retrieves from the filesystem. Should do so here as well.